### ISSUE TITLE
Speed up integration test and fix flakiness

### ISSUE THREAD

**Issue by ryardley** [gnosisguild/enclave#228]
Currently the integration tests take 14 minutes. We should run the tests in parallel and try to speed this up. We are getting flaky tests too that sometimes pass sometimes fail.

- [ ] #233 
- [x] Provide a way for any sleeps to be dependent on some kind of polling or check
- [x] Ensure all binaries are incrementally built ahead of time before the tests. (Binaries can sometimes take longer to build and the data submission window on the contract lapses)

**Comment by ryardley** [comment]
This was sped up to around 4 minutes so closing this. We can open a new issue if test speed becomes a problem again
